pkg/version/manager.go

```diff
@@ -23,19 +23,43 @@ import (
 	"github.com/Masterminds/semver/v3"
 
 	v1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
 	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
 	"k8c.io/kubermatic/v2/pkg/validation/nodeupdate"
 )
 
+type IncompatibilityCondition string
+
+const (
+	AlwaysCondition                IncompatibilityCondition = "always"
+	ExternalCloudProviderCondition IncompatibilityCondition = "externalCloudProvider"
```

Comment:
Please document these and the other new constants. I am not even sure what exactly these two control ^^